components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegate.kt

```diff
@@ -459,6 +459,31 @@ internal class GeckoPromptDelegate(private val geckoEngineSession: GeckoEngineSe
         return geckoResult
     }
 
+    override fun onRepostConfirmPrompt(
+        session: GeckoSession,
+        prompt: PromptDelegate.RepostConfirmPrompt
+    ): GeckoResult<PromptResponse>? {
+        val geckoResult = GeckoResult<PromptResponse>()
+
+        val onAccept: () -> Unit = {
+            geckoResult.complete(prompt.confirm(AllowOrDeny.ALLOW))
+        }
+        val onCancel: () -> Unit = {
+            geckoResult.complete(prompt.confirm(AllowOrDeny.DENY))
+            geckoEngineSession.notifyObservers { onRepostPromptCancelled() }
+        }
```

Comment:
Could you cover the `confirm` actions with an extra validation `!isComplete`?  Similar to what we are doing with [dismissSafely](https://github.com/mozilla-mobile/android-components/blob/00d48d45b9297f25e46faa886de73af08ec5ea8e/components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegate.kt#L583) , we have seen some issues when the app calls this functions after it has been called [before causing crashes](https://sentry.prod.mozaws.net/operations/firefox/?query=is%3Aunresolved+Cannot+confirm%2Fdismiss+a+Prompt+twice.).

Reply:
Sure! Thank you!

Reply:
Also added 2 more tests, one to check for this completed status, one to check that `onRepostPromptCancelled` is called.